src/shacl2code/lang/templates/golang.j2

```diff
@@ -0,0 +1,102 @@
+// package spdx_v3_0 provides structs for SPDX 3.0
+//
+// {{ disclaimer }}
+//
+// SPDX-License-Identifier: MIT
+
+{# helpful macros #}
+
+{%- macro array(prop) %}
+{%- if prop.max_count and prop.max_count >= 1 -%}
+[]
+{%- endif %}
+{%- endmacro %}
+
+{%- macro datatype(prop) %}
+{%- if prop.class_id -%}
+{{ export(classes.get(prop.class_id).clsname) }}
+{%- else -%}
+{{ DATATYPES[prop.datatype] }}
+{%- endif %}
+{%- endmacro %}
+
+{% macro interface(class) -%}
+type {{ export(class.clsname) }} interface {
+{%- for prop in class.properties %}
+	{{ export(prop.varname) }}() {{ array(prop) }}{{ datatype(prop) }}
+{%- endfor %}
+}
+{% endmacro %}
+
+{% macro struct(class) -%}
+type {{ export(class.clsname) }} struct {
+{%- for prop in class.properties %}
+}
+{%- endmacro %}
+
+package spdx_v3_0
```

Comment:
Instead of hardcoding, this should be specified as a command line argument in `GolangRender`